modules/navier_stokes/src/executioners/SIMPLE.C

```diff
@@ -0,0 +1,1279 @@
+  params.addParam<unsigned int>(
+      "passive_scalar_l_max_its",
+      10000,
+      "The maximum allowed iterations in the linear solver of the passive scalar equation.");
+
+  params.addParamNamesToGroup(
+      "momentum_l_tol momentum_l_abs_tol momentum_l_max_its pressure_l_tol pressure_l_abs_tol "
+      "pressure_l_max_its solid_energy_l_tol solid_energy_l_abs_tol solid_energy_l_max_its "
+      "energy_l_tol energy_l_abs_tol energy_l_max_its passive_scalar_l_tol "
+      "passive_scalar_l_abs_tol passive_scalar_l_max_its",
+      "Linear Iteration");
+
+  /*
+   * Pressure pin parameters for enclosed flows
+   */
+
+  params.addParam<bool>(
+      "pin_pressure", false, "If the pressure field needs to be pinned at a point.");
+  params.addParam<Real>(
+      "pressure_pin_value", 0, "The value which needs to be enforced for the pressure.");
+  params.addParam<Point>("pressure_pin_point", "The point where the pressure needs to be pinned.");
+
+  params.addParamNamesToGroup("pin_pressure pressure_pin_value pressure_pin_point", "Pressure Pin");
+
+  params.addParam<bool>(
+      "print_fields",
+      false,
+      "Use this to print the coupling and solution fields and matrices throughout the iteration.");
+
+  /*
+   * We suppress parameters which are not supported yet
+   */
+  params.suppressParameter<MooseEnum>("fixed_point_algorithm");
+  params.suppressParameter<unsigned int>("fixed_point_min_its");
+  params.suppressParameter<unsigned int>("fixed_point_max_its");
+  params.suppressParameter<bool>("accept_on_max_fixed_point_iteration");
+  params.suppressParameter<bool>("disable_fixed_point_residual_norm_check");
+  params.suppressParameter<Real>("fixed_point_rel_tol");
+  params.suppressParameter<Real>("fixed_point_abs_tol");
+  params.suppressParameter<unsigned int>("fixed_point_min_its");
+  params.suppressParameter<bool>("fixed_point_force_norms");
+  params.suppressParameter<PostprocessorName>("custom_pp");
+  params.suppressParameter<Real>("custom_rel_tol");
+  params.suppressParameter<Real>("custom_abs_tol");
+  params.suppressParameter<bool>("direct_pp_value");
+  params.suppressParameter<Real>("relaxation_factor");
+  params.suppressParameter<std::vector<std::string>>("transformed_variables");
+  params.suppressParameter<std::vector<PostprocessorName>>("transformed_postprocessors");
+  params.suppressParameter<std::vector<std::string>>("relaxed_variables");
+  params.suppressParameter<bool>("auto_advance");
+  params.suppressParameter<unsigned int>("max_xfem_update");
+  params.suppressParameter<bool>("update_xfem_at_timestep_begin");
+
+  return params;
+}
+
+SIMPLE::SIMPLE(const InputParameters & parameters)
+  : Executioner(parameters),
+    _problem(_fe_problem),
+    _time_step(_problem.timeStep()),
+    _time(_problem.time()),
+    _has_energy_system(isParamValid("energy_system")),
+    _has_solid_energy_system(_has_energy_system && isParamValid("solid_energy_system")),
```

Comment:
this wont complain if the solid energy system is set but the fluid energy is not. I think params will end up ignored

Reply:
added a paramerror